src/openloco/input.cpp

```diff
@@ -77,4 +83,786 @@ namespace openloco::input
+                _tooltipCursorX = x;
+                _tooltipCursorY = y;
+                scroll_begin(x, y, window, widget, widgetIndex);
+                break;
+
+            default:
+                if (window->is_enabled(widgetIndex) && !window->is_disabled(widgetIndex))
+                {
+                    audio::play_sound(audio::sound_id::click_1, window->x + ((widget->left + widget->right) / 2));
+
+                    // Set new cursor down widget
+                    _pressedWidgetIndex = widgetIndex;
+                    _pressedWindowType = window->type;
+                    _pressedWindowNumber = window->number;
+                    set_flag(input_flags::widget_pressed);
+                    _state = (uint8_t)input_state::widget_pressed;
+                    _clickRepeatTicks = 1;
+
+                    ui::windowmgr::invalidate_widget(window->type, window->number, widgetIndex);
+                    window->call_on_mouse_down(widgetIndex);
+                }
+
+                break;
+        }
+    }
+
+    // 0x004C85D1
+    static void window_resize_begin(int16_t x, int16_t y, ui::window* window, ui::widget_index widget_index)
+    {
+        _state = (uint8_t)input_state::resizing;
+        _pressedWidgetIndex = widget_index;
+        _dragLastX = x;
+        _dragLastY = y;
+        _dragWindowType = window->type;
+        _dragWindowNumber = window->number;
+        window->flags &= ~ui::window_flags::flag_15;
+    }
+
+    // 0x00C877D
+    static void window_position_begin(int16_t x, int16_t y, ui::window* window, ui::widget_index widget_index)
+    {
+        _state = (uint8_t)input_state::positioning_window;
+        _pressedWidgetIndex = widget_index;
+        _dragLastX = x;
+        _dragLastY = y;
+        _dragWindowType = window->type;
+        _dragWindowNumber = window->number;
+    }
+
+    static void state_normal_right(int16_t x, int16_t y, ui::window* window, ui::widget_t* widget, ui::widget_index widgetIndex)
+    {
+        ui::window_type windowType = ui::window_type::undefined;
+        ui::window_number windowNumber = 0;
+
+        if (window != nullptr)
+        {
+            windowType = window->type;
+            windowNumber = window->number;
+        }
+
+        ui::windowmgr::close(ui::window_type::tooltip);
+
+        // Window might have changed position in the list, therefore find it again
+        window = ui::windowmgr::find(windowType, windowNumber);
+        if (window == nullptr)
+        {
+            return;
+        }
+
+        window = ui::windowmgr::bring_to_front(window);
+
+        if (widgetIndex == -1)
+        {
+            return;
+        }
+
+        if (*modalWindowType != ui::window_type::undefined)
+        {
+            if (*modalWindowType == window->type)
+            {
+                scroll_begin(x, y, window, widget, widgetIndex);
+            }
+
+            return;
+        }
+
+        if (!is_title_mode())
```

Comment:
Stylistically, it might be worth changing this to return if `is_title_mode()` is true, rather than indenting an if-block here.